### ISSUE TITLE
Question about parameter fixing

### ISSUE THREAD

**Issue by c-box** [princeton-nlp/OptiPrompt#2]
I notice that you add the whole word embedding to the optimizer instead of just prompts?
```python
# Add word embeddings to the optimizer
optimizer = AdamW([{'params': model.base_model.embeddings.word_embeddings.parameters()}], lr=args.learning_rate, correct_bias=False)
```
Would the embedding of other words also be updated?

**Comment by a3616001** [comment]
Hi @c-box, thanks for your interest!

The embeddings of other words won't be updated, as we manually set the gradients of other word embeddings to be 0 [here](https://github.com/princeton-nlp/OptiPrompt/blob/main/code/run_optiprompt.py#L209).

**Comment by c-box** [comment]
Oh, I missed it. Thank you very much!
